release/__init__.py

```diff
@@ -122,6 +122,16 @@ def load_providers():
             for name in provider_names}
 
 
+def validate_cloudformation(template_url):
```

Comment:
This isn't something that the the "release" tool should know about (It really wants to interact with all the providers the same). We also need to be able to do cloudformation template validation of the advanced AWS templates which dcos_generate_config.sh is going to be outputting (#442)

Have two options to how to do this that I can see
1. Teach release to, after uploading artifacts, call a "validate" function for each artifact (better than nothing)
2. Try and get the templates back under the byte / complexity limit (my preference). There's probably a couple simple things we can move that will make these small enough (We really want as little as humanly possible in the cloud-config we put in the templates).